docs(router): update navigation event example to use event.code

portneon opened the pull request:
Update the code sample to use `event.code` instead of `event.reason` for cancellation checks, while preserving `event.reason` in the console.warn line. This aligns the example with the NavigationCancellationCode enum and prevents confusion.



## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/contributing-docs/commit-message-guidelines.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.dev application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: https://github.com/angular/angular/issues/65182


## What is the new behavior?


## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No





The previous example incorrectly compared event.code. Updated to use event.reason
to match the current NavigationCancel API. This fixes the type errors that were
caused by the old example.

Closes #65232
Closes #65235

SkyZeroZx commented on adev/src/content/guide/routing/lifecycle-and-events.md:
```diff
@@ -175,8 +176,8 @@ export class ErrorHandlerComponent {
       } else if (event instanceof NavigationCancel) {
-        console.warn('Navigation cancelled:', event.reason);
-        if (event.reason === NavigationCancellationCode.GuardRejected) {
+        console.warn('Navigation cancelled:', event.code);
```
Similar to what was mentioned earlier, I think `event.reason` should be preserved for the example on line 179

SkyZeroZx replied:
You can merge both commits into one.

portneon committed:
fix(router): use event.code instead of event.reason to fix type error
The example was comparing  (string) against  enum,
causing a TypeScript type error (TS2367). Also, the import  from
 was missing.

This change updates the example to use , ensuring type compatibility
and adding the missing import.

See: https://next.angular.dev/guide/routing/lifecycle-and-events#error-handling

portneon committed:
Merge branch 'angular:main' into fix/cla-single-commit
